packages/vue/src/T.ts

```diff
@@ -20,20 +21,36 @@ export const T = defineComponent({
     ns: { type: String as PropType<NsType> },
     language: { type: String as PropType<string> },
   },
-  setup() {
+  setup(props, context: SetupContext) {
+    const { slots } = context;
+    const { params } = props;
+    const slotsName = Object.keys(slots).filter((key) => key !== '_');
+    const slotsParams = {};
+    slotsName.forEach((key) => {
+      slotsParams[key] = `{${key}}`;
+    });
+
     const { t } = useTranslateInternal();
-    return { t };
+    const assignedParams = Object.assign({}, params, slotsParams);
+    return { t, assignedParams, slots, slotsName };
   },
   render() {
     const params: TranslateProps = {
       key: this.$props.keyName,
-      params: this.$props.params,
+      params: this.assignedParams,
       defaultValue: this.$props.defaultValue,
       noWrap: this.$props.noWrap,
       ns: this.$props.ns,
       language: this.$props.language,
     };
     const content = this.t(params);
-    return content;
+
+    if (this.slotsName.length === 0) return content;
+
+    const arrayOfChildren = convertStringToVNodeArrayWithSlots(
```

Comment:
You can do this by passing slots to params directly. The `t` function will actually do exactly what you are doing here (return array of strings and objects), so we can get rid of hacky `slotParams`

Reply:
yep, you are right

Reply:
@stepan662 I tried to pass slots directly to params, but we need to call the slot and pass the result. I adjusted the code. And now it works, but we still need to merge params and slots What do you think?